language/movie_nl/main.py

```diff
@@ -0,0 +1,367 @@
+
+        sentiment_writer.write(
+            to_sentiment_json(
+                document.doc_id,
+                sentiment_total,
+                document.label
+            )
+        )
+
+        sentiment_writer.write('\n')
+
+        for ent in entities:
+            ent_sent, frequency = collected_entities.get(ent, (0, 0))
+            ent_sent += sentiment_total
+            frequency += 1
+
+            collected_entities[ent] = (ent_sent, frequency)
+
+    for entity, e_tuple in collected_entities.items():
+        entity_writer.write(to_entity_json(entity, e_tuple))
+        entity_writer.write('\n')
+
+    sentiment_writer.flush()
+    entity_writer.flush()
+
+
+def document_generator(dir_path_pattern, count=None):
+    """Generator for the input movie documents."""
+    for running_count, item in enumerate(glob.iglob(dir_path_pattern)):
+        if count and running_count >= count:
+            raise StopIteration()
+
+        doc_id = os.path.basename(item)
+
+        with codecs.open(item, encoding='utf-8') as f:
+            try:
+                text = f.read()
+            except UnicodeDecodeError:
+                continue
+
+            yield Document(text, doc_id, item)
+
+
+def rank_entities(reader, sentiment=None, topn=None, reverse_bool=False):
+    """Rank the entities (actors) based on their sentiment
+    assigned from the movie."""
+
+    items = []
+    for item in reader:
+        json_item = json.loads(item)
+        sent = json_item.get('sentiment')
+        entity_item = (sent, json_item)
+
+        if sentiment:
+            if sentiment == 'pos' and sent > 0:
+                items.append(entity_item)
+            elif sentiment == 'neg' and sent < 0:
+                items.append(entity_item)
+        else:
+            items.append(entity_item)
+
+    items.sort(reverse=True)
+    items = [json.dumps(item[1]) for item in items]
+
+    if reverse_bool:
+        items.reverse()
+
+    if topn:
+        print('\n'.join(items[:topn]))
+    else:
+        print('\n'.join(items))
+
+
+def get_service():
+    """Build a client to the Google Cloud Natural Language API."""
+
+    credentials = GoogleCredentials.get_application_default()
+    scoped_credentials = credentials.create_scoped(
+        ['https://www.googleapis.com/auth/cloud-platform'])
+    http = httplib2.Http()
+    scoped_credentials.authorize(http)
+    return discovery.build('language', 'v1beta1', http=http)
+
+
+def analyze(input_dir, sentiment_writer, entity_writer, sample, log_file):
+    """Movie demo main program"""
+
+    # Create logger settings
+    logging.basicConfig(filename=log_file, level=logging.DEBUG)
+
+    # Create a Google Service object
+    service = get_service()
+
+    reader = document_generator(input_dir, sample)
+
+    # Process the movie documents
+    process_movie_reviews(service, reader, sentiment_writer, entity_writer)
+
+    # close reader and writers
+    sentiment_writer.close()
+    entity_writer.close()
```

Comment:
Ok I never noticed this change Jerjou made. I agree with you on that. Jerjou did you have other thoughts?